### PR TITLE
Fixes 4486,4487:  Modal rewrite/support uploaded content

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

- [x] [HMS-4486](https://issues.redhat.com/browse/HMS-4486): Update Create/Edit modal to support creating uploaded content
- [x] [HMS-4487](https://issues.redhat.com/browse/HMS-4487): Differentiate uploaded repositories in custom repo list/template list

Screenshot 2024-08-29 at 3 01 29 PM

Screenshot 2024-08-29 at 3 04 47 PM

New repository type option for Add modal: 
Screenshot 2024-08-29 at 3 11 52 PM
Screenshot 2024-08-29 at 3 06 28 PM
Screenshot 2024-08-29 at 3 05 12 PM
Screenshot 2024-08-29 at 3 06 09 PM

New repository type option for Edit modal: 
Screenshot 2024-08-29 at 3 12 21 PM
Screenshot 2024-08-29 at 3 12 36 PM

Additional information: 
- This completely rewrites the Add/edit content modals. Understandably, this will result in a number of QE test failures. 
I believe this will be a welcomed change once completed, as both from the dev/QE perspective this modal is now much more simple.

- Feedback on copy and design welcome and wanted! 🥂 🎉 

## Testing steps

- Test out creation of the different types of repositories.
- Regression testing is needed on both add/edit modals as they have been rewritten.
- Upload repositories will not show a URL and will instead show a label with icon and tooltip.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4486

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4487

File: src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx [hunk 1]
```diff
@@ -468,302 +373,261 @@ const AddContent = () => {
...
+          </FormGroup>
+          <FormGroup
+            label='Repository type'
+            fieldId='repositoryType'
+            hasNoPaddingTop
+            // labelIcon={
+            //   <Tooltip content='Put important explanation in here: Put important explanation in herePut important explanation in herePut important explanation in herePut important explanation in herePut important explanation in herePut important explanation in herePut important explanation in here'>
+            //     <OutlinedQuestionCircleIcon className='pf-u-ml-xs' color={global_Color_200.value} />
+            //   </Tooltip>
+            // }
```
Comment (jlsherrill) [inline src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx:408]: can this be removed?

File: src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx [hunk 2]
```diff
@@ -468,302 +373,261 @@ const AddContent = () => {
...
+          >
+            <Flex direction={{ default: 'column' }} gap={{ default: 'gap' }}>
+              <Hide hide={isEdit && contentOrigin === ContentOrigin.UPLOAD}>
+                <Radio
+                  isChecked={values.snapshot && values.origin === ContentOrigin.EXTERNAL}
+                  id='snapshot_radio'
+                  label='Snapshotting'
+                  description={
+                    values.snapshot && values.origin === ContentOrigin.EXTERNAL
+                      ? 'Enable snapshotting for an external repository, allowing you to build images with historical snapshots.'
```
Comment (jlsherrill) [inline src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx:418]: we should mention content templates

```Enable snapshotting for an external repository, allowing you to build images with historical snapshots and in content templates.```

File: src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx [hunk 3]
```diff
@@ -468,302 +373,261 @@ const AddContent = () => {
...
-                    createDataLengthOf1 ? classes.singleContentCol : classes.mainContentCol
+              <Hide hide={isInProd || (isEdit && contentOrigin === ContentOrigin.EXTERNAL)}>
+                <Radio
+                  isChecked={isUploadRepo}
+                  id='upload_radio'
+                  label='Upload'
+                  isDisabled={isEdit && contentOrigin === ContentOrigin.UPLOAD}
+                  description={
+                    isUploadRepo
+                      ? 'Create a repository to upload custom content to, snapshots will be taken after every new upload.'
```
Comment (jlsherrill) [inline src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx:449]: Mention the use caes:

Create a repository to upload custom content to, snapshots will be taken after every new upload and can be used in content templates.

File: src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx [hunk 4]
```diff
@@ -468,302 +373,261 @@ const AddContent = () => {
...
+                    setValues({ ...values, snapshot: true, origin: ContentOrigin.EXTERNAL })
+                  }
+                />
+                <Radio
+                  isChecked={!values.snapshot && values.origin === ContentOrigin.EXTERNAL}
+                  id='introspect_radio'
+                  label='Introspect only'
+                  description={
+                    !values.snapshot && values.origin === ContentOrigin.EXTERNAL
+                      ? 'Enable only introspection for an external repository, snapshots will not be taken.'
```
Comment (jlsherrill) [inline src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx:432]: Enable only introspection for an external repository, for building images against the external repository.

File: src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx (hunk 2 above)
Comment (jlsherrill) [reply src/Pages/Repositories/ContentListTable/components/AddContent/AddContent.tsx:418]: might want to say   "daily snapshotting"

**Review by xbhouse — APPROVED** [review]
looks good to me!

**Comment by Andrewgdewar** [comment]
Test [this](https://github.com/content-services/content-sources-frontend/pull/334) changes here are found there. 

The split button functionality for the add modal as well as the redirect after upload creation is also there.

**Comment by swadeley** [comment]
Hi, this needs IQE update because of the removal of the toggle switch and addition of the radio buttons.

**Comment by Andrewgdewar** [comment]
All changes seen her are within [this](https://github.com/content-services/content-sources-frontend/pull/334). 
So that I am not rebasing and updating two locations, I'll close this.

**Pull request closed without merging**